public/app/features/plugins/sql/datasource/SqlDatasource.ts

```diff
@@ -113,8 +113,15 @@ export abstract class SqlDatasource extends DataSourceWithBackend<SQLQuery, SQLO
     return expandedQueries;
   }
 
+  isCompleteQuery(query?: SQLQuery | undefined): boolean {
+    return query?.rawSql !== undefined && query.rawSql.trim() !== '';
+  }
+
   filterQuery(query: SQLQuery): boolean {
-    return !query.hide;
+    if (query.hide || !this.isCompleteQuery(query)) {
```

Comment:
Are all `filterQuery` implementations the same? Could we make it default implementation in the base class (`DataSourceWithBackend`)?

Reply:
They are not all technically the same, some datasources don't implement the hide logic - however, I do feel that is more of an oversight. Check out the latest push, is that what you're thinking?